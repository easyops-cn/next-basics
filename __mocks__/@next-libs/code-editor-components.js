import React from "react";

export function CodeEditorItem() {
  return <div>code editor</div>;
}
