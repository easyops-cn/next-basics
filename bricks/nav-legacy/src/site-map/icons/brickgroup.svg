<svg viewBox="0 0 11 12" xmlns="http://www.w3.org/2000/svg"><path d="M10.5 10.25a1.75 1.75 0 1 1-2.242-1.68V6.827L5.246 5.816l-3.01 1.01v1.742a1.75 1.75 0 1 1-.985.004V6.427l3.507-1.515V3.43A1.751 1.751 0 0 1 5.25 0a1.75 1.75 0 0 1 .492 3.43v1.49l3.504 1.503-.004 2.147a1.751 1.751 0 0 1 1.258 1.68zm-9.516 0a.766.766 0 1 0 1.533-.001.766.766 0 0 0-1.533.001zm7.766-.766a.766.766 0 1 0 .001 1.533.766.766 0 0 0-.001-1.533z" fill="#595959"/></svg>