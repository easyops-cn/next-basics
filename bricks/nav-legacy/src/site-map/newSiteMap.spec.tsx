import React from "react";
import {
  SubCategory,
  NewSiteMap,
  ModelTree,
  filterBySearch,
  matchSearchValue,
  RenderName,
  getNewFavouriteList,
  encode,
} from "./newSiteMap";
import { shallow, mount } from "enzyme";
import { Button, Popover } from "antd";
import { DownOutlined, UpOutlined } from "@ant-design/icons";
import { GeneralIcon, Link } from "@next-libs/basic-components";

import * as kit from "@next-core/brick-kit";

const data = {
  data: [
    {
      objectId: "APP",
      name: "app",
      category: "A",
    },
    {
      objectId: "HOST",
      name: "host",
      category: "B",
    },
  ],
};

(global as any).fetch = jest
  .fn()
  .mockImplementation(() => JSON.stringify({ data: data }));

describe("newSiteMap", () => {
  const groupList = [
    {
      name: "平台资源",
      title: "平台资源",
      objectList: [
        {
          name: "modelA",
          objectId: "A",
          category: "平台资源",
        },
      ],
      subCategory: [
        {
          name: "RS",
          title: "RS",
          objectList: [
            {
              name: "modelB",
              objectId: "B",
              category: "平台资源.RS",
            },
          ],
        },
        {
          name: "数据库变更",
          title: "数据库变更",
          objectList: [],
        },
      ],
    },
    {
      name: "应用资源",
      title: "应用资源",
      objectList: [{ name: "modelC", objectId: "C", category: "应用资源" }],
      subCategory: [
        {
          name: "门户",
          title: "门户",
          objectList: [
            {
              name: "modelD",
              objectId: "D",
              category: "应用资源.门户",
            },
          ],
        },
        {
          name: "服务监测",
          title: "服务监测",
          objectList: [],
        },
      ],
    },
  ];
  it("should work with SubCategory", () => {
    jest.spyOn(kit, "getRuntime").mockReturnValue({
      getFeatureFlags() {
        return {
          "cmdb-sitemap-menu-show-all-model": false,
        };
      },
    } as any);

    jest.spyOn(kit, "getHistory").mockReturnValue({ location: {} });
    const objectList = [
      { name: "集群", objectId: "CLUSTER" },
      { name: "测试", objectId: "TEST" },
    ];
    const wrapper = mount(
      <SubCategory title="门户" name="门户" objectList={objectList} />
    );
    expect(wrapper).toMatchSnapshot();
    const toggleBtn = wrapper.find(Button);
    expect(wrapper.find(DownOutlined)).toHaveLength(1);
    expect(wrapper.find(UpOutlined)).toHaveLength(0);
    expect(wrapper.find(Link)).toHaveLength(0);
    toggleBtn.simulate("click");
    expect(wrapper.find(DownOutlined)).toHaveLength(0);
    expect(wrapper.find(UpOutlined)).toHaveLength(1);
    expect(wrapper.find(Link)).toHaveLength(2);
  });
  it("should work with NewSiteMap", () => {
    const wrapper = mount(<NewSiteMap />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should work with ModelTree", () => {
    const wrapper = mount(<ModelTree groupList={groupList} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should work with matchSearchValue", () => {
    expect(matchSearchValue({ objectId: "APP", name: "应用" }, "app")).toBe(
      true
    );
    expect(matchSearchValue({ objectId: "APP", name: "应用" }, "应用2")).toBe(
      false
    );
  });
  it("should work with filterBySearch", () => {
    expect(filterBySearch(groupList, "modelC")).toEqual([
      {
        name: "应用资源",
        title: "应用资源",
        objectList: [{ name: "modelC", objectId: "C", category: "应用资源" }],
        subCategory: [],
      },
    ]);
    expect(filterBySearch(groupList, "")).toEqual(groupList);
  });
  it("function encode work", () => {
    expect(encode("]")).toEqual("\\]");
    expect(encode("*")).toEqual("\\*");
    expect(encode(")")).toEqual("\\)");
    expect(encode("$")).toEqual("\\$");
    expect(encode("?")).toEqual("\\?");
    expect(encode("}")).toEqual("\\}");
    expect(encode("+")).toEqual("\\+");
    expect(encode("^")).toEqual("\\^");
  });
});
describe("RenderName", () => {
  it("should work", () => {
    const wrapper = mount(<RenderName name="abc" />);
    expect(wrapper.find("span")).toHaveLength(1);
    expect(wrapper.text()).toEqual("abc");
  });
  it("should work", () => {
    const wrapper = mount(<RenderName name="abcABCDefg" highlightChar="bc" />);
    const splits = wrapper.find("span");
    expect(splits).toHaveLength(5);
    expect(splits.at(0).text()).toEqual("a");
    expect(splits.at(1).text()).toEqual("bc");
    expect(splits.at(2).text()).toEqual("A");
    expect(splits.at(3).text()).toEqual("BC");
    expect(splits.at(4).text()).toEqual("Defg");
    expect(splits.at(1).hasClass("highlightText")).toBe(true);
  });
});
describe("getNewFavouriteList", () => {
  expect(getNewFavouriteList("APP", ["APP", "HOST", "CLUSTER"], true)).toEqual([
    "APP",
    "HOST",
    "CLUSTER",
  ]);
  expect(getNewFavouriteList("APP", ["APP", "HOST", "CLUSTER"], false)).toEqual(
    ["HOST", "CLUSTER"]
  );
  expect(
    getNewFavouriteList("AGENT", ["APP", "HOST", "CLUSTER"], true)
  ).toEqual(["APP", "HOST", "CLUSTER", "AGENT"]);
});
