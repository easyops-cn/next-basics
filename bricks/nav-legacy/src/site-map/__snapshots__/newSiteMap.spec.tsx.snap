// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newSiteMap should work with ModelTree 1`] = `
<ModelTree
  groupList={
    Array [
      Object {
        "name": "平台资源",
        "objectList": Array [
          Object {
            "category": "平台资源",
            "name": "modelA",
            "objectId": "A",
          },
        ],
        "subCategory": Array [
          Object {
            "name": "RS",
            "objectList": Array [
              Object {
                "category": "平台资源.RS",
                "name": "modelB",
                "objectId": "B",
              },
            ],
            "title": "RS",
          },
          Object {
            "name": "数据库变更",
            "objectList": Array [],
            "title": "数据库变更",
          },
        ],
        "title": "平台资源",
      },
      Object {
        "name": "应用资源",
        "objectList": Array [
          Object {
            "category": "应用资源",
            "name": "modelC",
            "objectId": "C",
          },
        ],
        "subCategory": Array [
          Object {
            "name": "门户",
            "objectList": Array [
              Object {
                "category": "应用资源.门户",
                "name": "modelD",
                "objectId": "D",
              },
            ],
            "title": "门户",
          },
          Object {
            "name": "服务监测",
            "objectList": Array [],
            "title": "服务监测",
          },
        ],
        "title": "应用资源",
      },
    ]
  }
>
  <div
    className="masonry"
  >
    <div
      className="column"
      key="0"
    >
      <div
        className="newItem"
      >
        <div
          className="subCategoryContainer"
        >
          <GeneralIcon
            icon={
              Object {
                "category": "app",
                "icon": "workspace",
                "lib": "easyops",
              }
            }
          >
            <AntdIcon
              className=""
              component={[Function]}
              data-icon="workspace"
            >
              <span
                className="anticon"
                data-icon="workspace"
                role="img"
              >
                <component
                  aria-hidden="true"
                  className=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  <BrickIcon
                    category="app"
                    icon="workspace"
                  >
                    <span
                      className="easyops-icon"
                    />
                  </BrickIcon>
                </component>
              </span>
            </AntdIcon>
          </GeneralIcon>
          <span
            className="categoryTitle"
          >
            平台资源
          </span>
        </div>
        <SubCategory
          key="0"
          name="RS"
          objectList={
            Array [
              Object {
                "category": "平台资源.RS",
                "name": "modelB",
                "objectId": "B",
              },
            ]
          }
          title="RS"
        >
          <div
            className="subCategoryItemContainer"
          >
            <Button
              onClick={[Function]}
              type="link"
            >
              <button
                className="ant-btn ant-btn-link"
                onClick={[Function]}
                type="button"
              >
                <LoadingIcon
                  existIcon={false}
                  loading={false}
                  prefixCls="ant-btn"
                >
                  <CSSMotion
                    motionName="ant-btn-loading-icon-motion"
                    onAppearActive={[Function]}
                    onAppearStart={[Function]}
                    onEnterActive={[Function]}
                    onEnterStart={[Function]}
                    onLeaveActive={[Function]}
                    onLeaveStart={[Function]}
                    removeOnLeave={true}
                    visible={false}
                  >
                    <DomWrapper />
                  </CSSMotion>
                </LoadingIcon>
                <div
                  className="toggleWrapper"
                  key=".0"
                >
                  <span
                    className="categoryNameContainer"
                  >
                    RS
                  </span>
                  <ForwardRef(DownOutlined)>
                    <AntdIcon
                      icon={
                        Object {
                          "icon": Object {
                            "attrs": Object {
                              "focusable": "false",
                              "viewBox": "64 64 896 896",
                            },
                            "children": Array [
                              Object {
                                "attrs": Object {
                                  "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                                },
                                "tag": "path",
                              },
                            ],
                            "tag": "svg",
                          },
                          "name": "down",
                          "theme": "outlined",
                        }
                      }
                    >
                      <span
                        aria-label="down"
                        className="anticon anticon-down"
                        role="img"
                      >
                        <IconReact
                          icon={
                            Object {
                              "icon": Object {
                                "attrs": Object {
                                  "focusable": "false",
                                  "viewBox": "64 64 896 896",
                                },
                                "children": Array [
                                  Object {
                                    "attrs": Object {
                                      "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                                    },
                                    "tag": "path",
                                  },
                                ],
                                "tag": "svg",
                              },
                              "name": "down",
                              "theme": "outlined",
                            }
                          }
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="down"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            key="svg-down"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              key="svg-down-svg-0"
                            />
                          </svg>
                        </IconReact>
                      </span>
                    </AntdIcon>
                  </ForwardRef(DownOutlined)>
                </div>
              </button>
            </Button>
          </div>
        </SubCategory>
        <div
          style={
            Object {
              "height": "5px",
            }
          }
        />
        <div
          className="objectItemContainer categoryNameContainer objectNameContainer"
          key="0"
        >
          <Link>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <RenderName
                name="modelA"
              >
                <span>
                  modelA
                </span>
              </RenderName>
            </a>
          </Link>
          <Button
            onClick={[Function]}
            type="text"
          >
            <button
              className="ant-btn ant-btn-text"
              onClick={[Function]}
              type="button"
            >
              <LoadingIcon
                existIcon={false}
                loading={false}
                prefixCls="ant-btn"
              >
                <CSSMotion
                  motionName="ant-btn-loading-icon-motion"
                  onAppearActive={[Function]}
                  onAppearStart={[Function]}
                  onEnterActive={[Function]}
                  onEnterStart={[Function]}
                  onLeaveActive={[Function]}
                  onLeaveStart={[Function]}
                  removeOnLeave={true}
                  visible={false}
                >
                  <DomWrapper />
                </CSSMotion>
              </LoadingIcon>
              <ForwardRef(StarOutlined)
                className="collectStar"
                key=".0"
              >
                <AntdIcon
                  className="collectStar"
                  icon={
                    Object {
                      "icon": Object {
                        "attrs": Object {
                          "focusable": "false",
                          "viewBox": "64 64 896 896",
                        },
                        "children": Array [
                          Object {
                            "attrs": Object {
                              "d": "M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z",
                            },
                            "tag": "path",
                          },
                        ],
                        "tag": "svg",
                      },
                      "name": "star",
                      "theme": "outlined",
                    }
                  }
                >
                  <span
                    aria-label="star"
                    className="anticon anticon-star collectStar"
                    role="img"
                  >
                    <IconReact
                      icon={
                        Object {
                          "icon": Object {
                            "attrs": Object {
                              "focusable": "false",
                              "viewBox": "64 64 896 896",
                            },
                            "children": Array [
                              Object {
                                "attrs": Object {
                                  "d": "M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z",
                                },
                                "tag": "path",
                              },
                            ],
                            "tag": "svg",
                          },
                          "name": "star",
                          "theme": "outlined",
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        key="svg-star"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                          key="svg-star-svg-0"
                        />
                      </svg>
                    </IconReact>
                  </span>
                </AntdIcon>
              </ForwardRef(StarOutlined)>
            </button>
          </Button>
        </div>
      </div>
    </div>
    <div
      className="column"
      key="1"
    >
      <div
        className="newItem"
      >
        <div
          className="subCategoryContainer"
        >
          <GeneralIcon
            icon={
              Object {
                "icon": "cube",
                "lib": "fa",
                "prefix": "fas",
              }
            }
          >
            <AntdIcon
              component={[Function]}
              data-icon="cube"
              style={
                Object {
                  "verticalAlign": 0,
                }
              }
            >
              <span
                className="anticon"
                data-icon="cube"
                role="img"
                style={
                  Object {
                    "verticalAlign": 0,
                  }
                }
              >
                <component
                  aria-hidden="true"
                  className=""
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  width="1em"
                >
                  <mockConstructor
                    className="faIcon"
                    icon={
                      Array [
                        "fas",
                        "cube",
                      ]
                    }
                  >
                    <div>
                      fas
                      cube
                    </div>
                  </mockConstructor>
                </component>
              </span>
            </AntdIcon>
          </GeneralIcon>
          <span
            className="categoryTitle"
          >
            应用资源
          </span>
        </div>
        <SubCategory
          key="0"
          name="门户"
          objectList={
            Array [
              Object {
                "category": "应用资源.门户",
                "name": "modelD",
                "objectId": "D",
              },
            ]
          }
          title="门户"
        >
          <div
            className="subCategoryItemContainer"
          >
            <Button
              onClick={[Function]}
              type="link"
            >
              <button
                className="ant-btn ant-btn-link"
                onClick={[Function]}
                type="button"
              >
                <LoadingIcon
                  existIcon={false}
                  loading={false}
                  prefixCls="ant-btn"
                >
                  <CSSMotion
                    motionName="ant-btn-loading-icon-motion"
                    onAppearActive={[Function]}
                    onAppearStart={[Function]}
                    onEnterActive={[Function]}
                    onEnterStart={[Function]}
                    onLeaveActive={[Function]}
                    onLeaveStart={[Function]}
                    removeOnLeave={true}
                    visible={false}
                  >
                    <DomWrapper />
                  </CSSMotion>
                </LoadingIcon>
                <div
                  className="toggleWrapper"
                  key=".0"
                >
                  <span
                    className="categoryNameContainer"
                  >
                    门户
                  </span>
                  <ForwardRef(DownOutlined)>
                    <AntdIcon
                      icon={
                        Object {
                          "icon": Object {
                            "attrs": Object {
                              "focusable": "false",
                              "viewBox": "64 64 896 896",
                            },
                            "children": Array [
                              Object {
                                "attrs": Object {
                                  "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                                },
                                "tag": "path",
                              },
                            ],
                            "tag": "svg",
                          },
                          "name": "down",
                          "theme": "outlined",
                        }
                      }
                    >
                      <span
                        aria-label="down"
                        className="anticon anticon-down"
                        role="img"
                      >
                        <IconReact
                          icon={
                            Object {
                              "icon": Object {
                                "attrs": Object {
                                  "focusable": "false",
                                  "viewBox": "64 64 896 896",
                                },
                                "children": Array [
                                  Object {
                                    "attrs": Object {
                                      "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                                    },
                                    "tag": "path",
                                  },
                                ],
                                "tag": "svg",
                              },
                              "name": "down",
                              "theme": "outlined",
                            }
                          }
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="down"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            key="svg-down"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              key="svg-down-svg-0"
                            />
                          </svg>
                        </IconReact>
                      </span>
                    </AntdIcon>
                  </ForwardRef(DownOutlined)>
                </div>
              </button>
            </Button>
          </div>
        </SubCategory>
        <div
          style={
            Object {
              "height": "5px",
            }
          }
        />
        <div
          className="objectItemContainer categoryNameContainer objectNameContainer"
          key="0"
        >
          <Link>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <RenderName
                name="modelC"
              >
                <span>
                  modelC
                </span>
              </RenderName>
            </a>
          </Link>
          <Button
            onClick={[Function]}
            type="text"
          >
            <button
              className="ant-btn ant-btn-text"
              onClick={[Function]}
              type="button"
            >
              <LoadingIcon
                existIcon={false}
                loading={false}
                prefixCls="ant-btn"
              >
                <CSSMotion
                  motionName="ant-btn-loading-icon-motion"
                  onAppearActive={[Function]}
                  onAppearStart={[Function]}
                  onEnterActive={[Function]}
                  onEnterStart={[Function]}
                  onLeaveActive={[Function]}
                  onLeaveStart={[Function]}
                  removeOnLeave={true}
                  visible={false}
                >
                  <DomWrapper />
                </CSSMotion>
              </LoadingIcon>
              <ForwardRef(StarOutlined)
                className="collectStar"
                key=".0"
              >
                <AntdIcon
                  className="collectStar"
                  icon={
                    Object {
                      "icon": Object {
                        "attrs": Object {
                          "focusable": "false",
                          "viewBox": "64 64 896 896",
                        },
                        "children": Array [
                          Object {
                            "attrs": Object {
                              "d": "M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z",
                            },
                            "tag": "path",
                          },
                        ],
                        "tag": "svg",
                      },
                      "name": "star",
                      "theme": "outlined",
                    }
                  }
                >
                  <span
                    aria-label="star"
                    className="anticon anticon-star collectStar"
                    role="img"
                  >
                    <IconReact
                      icon={
                        Object {
                          "icon": Object {
                            "attrs": Object {
                              "focusable": "false",
                              "viewBox": "64 64 896 896",
                            },
                            "children": Array [
                              Object {
                                "attrs": Object {
                                  "d": "M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z",
                                },
                                "tag": "path",
                              },
                            ],
                            "tag": "svg",
                          },
                          "name": "star",
                          "theme": "outlined",
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        key="svg-star"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                          key="svg-star-svg-0"
                        />
                      </svg>
                    </IconReact>
                  </span>
                </AntdIcon>
              </ForwardRef(StarOutlined)>
            </button>
          </Button>
        </div>
      </div>
    </div>
  </div>
</ModelTree>
`;

exports[`newSiteMap should work with NewSiteMap 1`] = `
<NewSiteMap>
  <div
    className="menuPopover"
    id="menuPopover"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="newSiteMap"
    >
      <span
        className="appName"
      >
        IT_RESOURCE_MANAGEMENT
      </span>
    </div>
    <div
      className="sitMapModelContainer"
      id="sitMapModelContainer"
    >
      <DrawerWrapper
        className="popoverInMenu"
        getContainer={[Function]}
        height="calc(60vh - 24px)"
        maskClosable={true}
        onClose={[Function]}
        placement="top"
        title={null}
        visible={false}
        zIndex={-1}
      >
        <Drawer
          className="popoverInMenu"
          getContainer={[Function]}
          height="calc(60vh - 24px)"
          maskClosable={true}
          onClose={[Function]}
          placement="top"
          prefixCls="ant-drawer"
          title={null}
          visible={false}
          zIndex={-1}
        >
          <DrawerWrapper
            afterVisibleChange={[Function]}
            className="popoverInMenu"
            defaultOpen={false}
            duration=".3s"
            ease="cubic-bezier(0.78, 0.14, 0.15, 0.86)"
            forceRender={false}
            getContainer={[Function]}
            handler={false}
            height="calc(60vh - 24px)"
            keyboard={true}
            level={null}
            maskClosable={true}
            maskStyle={Object {}}
            onChange={[Function]}
            onClose={[Function]}
            open={false}
            placement="top"
            prefixCls="ant-drawer"
            showMask={true}
            style={
              Object {
                "transform": undefined,
                "zIndex": -1,
              }
            }
            wrapperClassName=""
          >
            <PortalWrapper
              forceRender={false}
              getContainer={[Function]}
              visible={false}
              wrapperClassName=""
            />
          </DrawerWrapper>
        </Drawer>
      </DrawerWrapper>
    </div>
  </div>
</NewSiteMap>
`;

exports[`newSiteMap should work with SubCategory 1`] = `
<SubCategory
  name="门户"
  objectList={
    Array [
      Object {
        "name": "集群",
        "objectId": "CLUSTER",
      },
      Object {
        "name": "测试",
        "objectId": "TEST",
      },
    ]
  }
  title="门户"
>
  <div
    className="subCategoryItemContainer"
  >
    <Button
      onClick={[Function]}
      type="link"
    >
      <button
        className="ant-btn ant-btn-link"
        onClick={[Function]}
        type="button"
      >
        <LoadingIcon
          existIcon={false}
          loading={false}
          prefixCls="ant-btn"
        >
          <CSSMotion
            motionName="ant-btn-loading-icon-motion"
            onAppearActive={[Function]}
            onAppearStart={[Function]}
            onEnterActive={[Function]}
            onEnterStart={[Function]}
            onLeaveActive={[Function]}
            onLeaveStart={[Function]}
            removeOnLeave={true}
            visible={false}
          >
            <DomWrapper />
          </CSSMotion>
        </LoadingIcon>
        <div
          className="toggleWrapper"
          key=".0"
        >
          <span
            className="categoryNameContainer"
          >
            门户
          </span>
          <ForwardRef(DownOutlined)>
            <AntdIcon
              icon={
                Object {
                  "icon": Object {
                    "attrs": Object {
                      "focusable": "false",
                      "viewBox": "64 64 896 896",
                    },
                    "children": Array [
                      Object {
                        "attrs": Object {
                          "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                        },
                        "tag": "path",
                      },
                    ],
                    "tag": "svg",
                  },
                  "name": "down",
                  "theme": "outlined",
                }
              }
            >
              <span
                aria-label="down"
                className="anticon anticon-down"
                role="img"
              >
                <IconReact
                  icon={
                    Object {
                      "icon": Object {
                        "attrs": Object {
                          "focusable": "false",
                          "viewBox": "64 64 896 896",
                        },
                        "children": Array [
                          Object {
                            "attrs": Object {
                              "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                            },
                            "tag": "path",
                          },
                        ],
                        "tag": "svg",
                      },
                      "name": "down",
                      "theme": "outlined",
                    }
                  }
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    key="svg-down"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      key="svg-down-svg-0"
                    />
                  </svg>
                </IconReact>
              </span>
            </AntdIcon>
          </ForwardRef(DownOutlined)>
        </div>
      </button>
    </Button>
  </div>
</SubCategory>
`;
